### PR TITLE
Add toolbar namespace, image selection and image shapes

### PR DESCRIPTION
**Pull request by fdardenne**

### CODE REVIEW

File: addons/html_editor/static/src/main/image_plugin.js [hunk 1]
```diff
@@ -3,15 +3,98 @@ import { Plugin } from "../plugin";
...
+
+    setup() {
+        this.addDomListener(this.editable, "mouseup", (e) => {
+            if (e.target.tagName === "IMG") {
+                const selection = this.document.getSelection();
+                const range = this.document.createRange();
+                range.selectNode(e.target);
+                selection.removeAllRanges();
+                selection.addRange(range);
+            }
```
Comment (FrancoisGe) [inline addons/html_editor/static/src/main/image_plugin.js]: You cant use selection plugin ?

File: addons/html_editor/static/src/main/toolbar/toolbar.js [hunk 2]
```diff
@@ -12,15 +12,29 @@ export class Toolbar extends Component {
...
                 buttonGroups: Array,
                 buttonsActiveState: Object,
+                namespace: Object,
             },
         },
     };
 
     setup() {
         this.buttonsActiveState = useState(this.props.toolbar.buttonsActiveState);
+        this.namespace = useState(this.props.toolbar.namespace);
```
Comment (FrancoisGe) [inline addons/html_editor/static/src/main/toolbar/toolbar.js]: namespace could change during the life of the toolbar ?

File: addons/html_editor/static/src/main/toolbar/toolbar.js [hunk 3]
```diff
@@ -12,15 +12,29 @@ export class Toolbar extends Component {
...
     }
 
     dispatch(cmd, payload) {
         this.props.toolbar.dispatch(cmd, payload);
     }
+
+    getFilteredButtonGroups() {
+        if (this.namespace.tag) {
```
Comment (FrancoisGe) [inline addons/html_editor/static/src/main/toolbar/toolbar.js]: why namespace need tag ? namespace in command => string, namespce from props => { tag } look confusing

File: addons/html_editor/static/src/main/toolbar/toolbar.js (hunk 2 above)
Comment (fdardenne) [reply addons/html_editor/static/src/main/toolbar/toolbar.js]: Yes. The namespace tag change when the selection change

File: addons/html_editor/static/src/main/toolbar/toolbar.js (hunk 3 above)
Comment (fdardenne) [reply addons/html_editor/static/src/main/toolbar/toolbar.js]: in `toolbar_plugin.js` the namespace is defined as an object with the tag key:
```js
this.namespace = reactive({ tag: undefined });
```
in the `Toolbar` component it is passed as an object with the tag key.

The object with `tag` is not convenient but the problem is that we need a Set or a Map we can mutate to not lose the proxy. If you've any suggestion I'm open

**Pull request merged**